test/cmd/apply.sh

```diff
@@ -438,6 +438,87 @@ run_kubectl_server_side_apply_tests() {
   # clean-up
   kubectl delete -f hack/testdata/pod.yaml "${kube_flags[@]:?}"
 
+    # Test apply migration
+
+  # Create a configmap in the cluster with client-side apply:
+  output_message=$(cat <<EOF | kubectl apply --server-side=false -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test
+data:
+  key: value
+  legacy: unused
+EOF
+  )
+
+  kube::test::if_has_string "${output_message}" 'configmap/test created'
+
+  # Apply the same manifest with --server-side flag, as per server-side-apply migration instructions:
+  output_message=$(cat <<EOF | kubectl apply --server-side -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test
+data:
+  key: value
+  legacy: unused
+EOF
+  )
+
+  kube::test::if_has_string "${output_message}" 'configmap/test serverside-applied'
+
+  # Apply the object a third time using server-side-apply, but this time removing
+  # a field. Old versions of kubectl would not allow the field to be removed"
+  output_message=$(cat <<EOF | kubectl apply --server-side -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test
+data:
+  key: value
+EOF
+  )
+
+  kube::test::if_has_string "${output_message}" 'configmap/test serverside-applied'
+
+  # Fetch the object and check to see that it does not have a field 'legacy'
+  kube::test::get_object_assert "configmap test" "{{ .data.legacy }}" '<no value>'
+
+  # CSA the object after it has been server-side-applied and had a field removed
+  # Add new key with client-side-apply.
```

Comment:
Could we have this part of the test attempt a deletion as well, to prove that the downgrade is working properly? (or do we already have one in the suite that would fail without the annotation retention code?)

Reply:
Am I understanding correctly that you are requesting here to remove  a field in the test, similar to https://github.com/kubernetes/kubernetes/pull/112905#discussion_r1010993799 ?

Reply:
Yeah, it's the same request. I was doing my review in a tab I hadn't refreshed in hours. 🤦‍♀️

Reply:
copying comment here too:

I added a component to the integration test where client-side-apply removes a key added by server-side-apply.